scripts/eosio_build_darwin.sh

```diff
@@ -1,406 +1,271 @@
+	printf "\\n${DISPLAY}\\n"
+	echo "Do you wish to install these packages?"
+	select yn in "Yes" "No"; do
+		case $yn in
+			[Yy]* )
+				"${XCODESELECT}" --install 2>/dev/null;
+				printf "\\nDo you wish to update homebrew packages first?\\n"
+				select yn in "Yes" "No"; do
+					case $yn in
+						[Yy]* ) 
+							printf "\\n\\nUpdating...\\n\\n"
+							if ! brew update; then
+								printf "\\nbrew update failed.\\n"
+								printf "\\nExiting now.\\n\\n"
 								exit 1;
+							else
+								printf "\\brew update complete.\\n"
 							fi
-						fi
-					break;;
-					[Nn]* ) echo "User cancelled installation of Boost libraries, Exiting now."; exit;;
-					* ) echo "Please type 1 for yes or 2 for no.";;
-				esac
-			done
-		fi
-		printf "\\tInstalling boost libraries.\\n"
-		if ! "${BREW}" install https://raw.githubusercontent.com/Homebrew/homebrew-core/f946d12e295c8a27519b73cc810d06593270a07f/Formula/boost.rb
-		then
-			printf "\\tUnable to install boost 1.67 libraries at this time. 0\\n"
-			printf "\\tExiting now.\\n\\n"
-			exit 1;
-		fi
-		if [ -d "$BUILD_DIR" ]; then
-			if ! rm -rf "$BUILD_DIR"
-			then
-			printf "\\tUnable to remove directory %s. Please remove this directory and run this script %s again. 0\\n" "$BUILD_DIR" "${BASH_SOURCE[0]}"
-			printf "\\tExiting now.\\n\\n"
-			exit 1;
-			fi
-		fi
-		printf "\\tBoost 1.67.0 successfully installed @ /usr/local.\\n"
-	else
-		printf "\\tBoost 1.67.0 found at /usr/local.\\n"
-	fi
+						break;;
+						[Nn]* ) echo "Proceeding without update!"
+						break;;
+						* ) echo "Please type 1 for yes or 2 for no.";;
+					esac
+				done
 
-	printf "\\n\\tChecking MongoDB C++ driver installation.\\n"
-	MONGO_INSTALL=true
+				brew tap eosio/eosio # Required to install mongo-cxx-driver with static library
+				printf "\\nInstalling Dependencies.\\n"
+				# Ignore cmake so we don't install a newer version.
+				# Build from source to use local cmake
+				# DON'T INSTALL llvm@4 WITH --force!
+				OIFS="$IFS"
+				IFS=$','
+				for DEP in $DEPS; do
+					# Eval to support string/arguments with $DEP
+					if ! eval $BREW install $DEP; then
+						printf "Homebrew exited with the above errors.\\n"
+						printf "Exiting now.\\n\\n"
+						exit 1;
+					fi
+				done
+				IFS="$OIFS"
+			break;;
+			[Nn]* ) echo "User aborting installation of required dependencies, Exiting now."; exit;;
+			* ) echo "Please type 1 for yes or 2 for no.";;
+		esac
+	done
+else
+	printf "No required Home Brew dependencies to install.\\n"
+fi
 
-    if [ -e "/usr/local/lib/libmongocxx-static.a" ]; then
-		MONGO_INSTALL=false
-		if ! version=$( grep "Version:" /usr/local/lib/pkgconfig/libmongocxx-static.pc | tr -s ' ' | awk '{print $2}' )
-		then
-			printf "\\tUnable to determine mongodb-cxx-driver version.\\n"
-			printf "\\tExiting now.\\n\\n"
-			exit 1;
-		fi
+# Failed to find Gettext libintl (missing: Intl_INCLUDE_DIR); known bug
+brew unlink gettext 2> /dev/null && brew link --force gettext
```

Comment:
I recommend not force linking anything.

Reply:
I agree, but the problem requires the --force flag to fix. I wish we didn't need the workaround at all really.

Reply:
What is the problem? Actually where is gettext used? AFAIK not for nodeos/cleos/keosd

Reply:
Looks like gettext is a dependency of libidn2, which is required by wget. I believe I removed the majority of reliance on wget to download source packages (using Curl instead), so maybe we can remove it?

Reply:
Even if it is a dependency of wget, it doesn't need to be force linked to use wget.

Reply:
Looks like gettext is needed for cleos, actually. Surprised me